rfcs/2022-1-10-design-tokens.md

```diff
@@ -0,0 +1,66 @@
+# RFC: Implementation of design tokens
+
+**Date** : Jan 10, 2022
+**Status** : Review
+
+## Problem
+
+Presently, the inconsistency of design rules between the DSM, Designers making and consuming changes, and the design system codebase lead to **confusion**, **duplicated work**, **inconsistency** and **disorganization** within all three systems.
+
+- **Designers** should have an organized set of elements which standardize their process and add consistency to their production.
+- **Stakeholders and Project Managers** could be better served when discussing visual changes if there were a library of standards for them to choose from when making decisions.
+- **Developers** could eliminate confusion and the hacking together of solutions when presented with a visual change by utilizing the same set of elements as everyone else.
+
+## Proposal
+
+[Design tokens](https://www.invisionapp.com/inside-design/design-tokens/) can help us solve this problem.
+
+> Design tokens are all the values needed to construct and maintain a design system — spacing, color, typography, object styles, animation, etc. — represented as data. These can represent anything defined by design: a color as a RGB value, an opacity as a number, an animation ease as Bezier coordinates. They’re used in place of hard-coded values in order to ensure flexibility and unity across all product experiences. - [Adobe Spectrum](https://spectrum.adobe.com/page/design-tokens/)
+
+Design tokens can be codified in many ways using a variety of data storage formats. Since the CMSWDS is implemented in Javascript/Typescript, we can utilize the modularity and type safety of Typescript to manage the current set of styles.
+
+A design token solution is needed in which:
+
+1. The set of design tokens remain finite and equal
+2. The set of design tokens maintain consistency between application (DSM, Sketch, React and other future platforms such as iOS, Android, Web Components)
```

Comment:
Maybe we mention net-new design systems, as well as external design systems we inherit? Those are probably more likely than iOS/Android app work in the short term.